jurity/recommenders/interlist_diversity.py

```diff
@@ -0,0 +1,278 @@
+                        working_memory: int = None) -> Tuple[float, int]:
+    """
+    Inter-List Diversity@k measures the inter-list diversity of the recommendations when only k recommendations are
+    made to the user. It measures how user's lists of recommendations are different from each other. This metric has a
+    range in :math:`[0, 1]`. The higher this metric is, the more diversified lists of items are recommended to different
+    users. Let :math:`U` denote the set of :math:`N` unique users, :math:`u_i`, :math:`u_j \in U` denote the i-th and
+    j-th user in the user set, :math:`i, j \in \{0,1,\cdots,N\}`. :math:`R_{u_i}` is the binary indicator vector
+    representing provided recommendations for :math:`u_i`. :math:`I` is the set of all unique user pairs,
+    :math:`\\forall~i<j, \{u_i, u_j\} \in I`.
+
+    .. math::
+            Inter \mbox{-} list~diversity = \\frac{\sum_{i,j, \{u_i, u_j\} \in I}(cosine\_distance(R_{u_i}, R_{u_j}))}{|I|}
+
+    Parameters
+    ----------
+    predicted_results: pd.DataFrame
+        Recommendations data frame with (user_id, item_id, score) in each row.
+    k: int
+        Top-k recommendations to consider.
+    user_id_column: str
+        User id column name.
+    item_id_column: str
+        Item id column name.
+    click_column: str
+        Recommendation score column name.
+    user_sample_size: Union[int, float, None]
+        When input is an integer, it defines the number of randomly sampled users. When input is float, it defines the
+        proportion of users to randomly sample for evaluation. If it is None, all users are included. Default=10,000.
+    seed: int
+        The seed used to create random state.
+    metric: Union[str, Callable]
+        Default = 'cosine'. The distance metric leveraged by sklearn.metrics.pairwise_distances_chunked.
+        The metric to use when calculating distance between instances in a feature array.
+        If metric is a string, it must be one of the options allowed by scipy.spatial.distance.pdist for its metric
+        parameter, or a metric listed in pairwise.PAIRWISE_DISTANCE_FUNCTIONS. If metric is a callable function,
+        it is called on each pair of instances (rows) and the resulting value recorded.
+        The callable should take two arrays from X as input and return a value indicating the distance between them.
+    num_runs: int
+        num_runs is used to report the approximation of Inter-List Diversity over multiple runs on smaller
+        samples of users, default=10, for a speed-up on evaluations. The sampling size is defined by
+        user_sample_size. The final result is averaged over the multiple runs.
+    n_jobs: int
+        Number of jobs to use for computation in parallel, leveraged by sklearn.metrics.pairwise_distances_chunked.
+        -1 means using all processors. Default=1.
+    working_memory: Union[int, None]
```

Comment:
add an example of what input to `working_memory` should be if not none, I assume its `1024` and not `1024MB`?